File: tree.py
class Node:
    def __init__(self, operator, value, left, right):
        self.operator = operator
        self.value = value
        self.left = left
        self.right = right

    def result(self):
        if self.operator == "x":
            return self.left.result() * self.right.result()
        elif self.operator == "÷":
            return self.left.result() / self.right.result()
        elif self.operator == "+":
            return self.left.result() + self.right.result()
        elif self.operator == "-":
            return self.left.result() - self.right.result()
        else:
            return float(self.value)

    def __str__(self):
        if self.operator == "x":
            return f"({self.left} x {self.right})"
        elif self.operator == "÷":
            return f"({self.left} ÷ {self.right})"
        elif self.operator == "+":
            return f"({self.left} + {self.right})"
        elif self.operator == "-":
            return f"({self.left} - {self.right})"
        else:
            return str(self.value)

import unittest

class TestNode(unittest.TestCase):
    def test_node(self):
        tree = Node(
            "÷",
            None,
            Node(
                "+",
                None,
                Node("", 7, None, None),
                Node(
                    "x",
                    None,
                    Node("-", None, Node("", 3, None, None), Node("", 2, None, None)),
                    Node("", 5, None, None)
                )
            ),
            Node("", 6, None, None)
        )
        self.assertEqual(str(tree), "((7 + ((3 - 2) x 5)) ÷ 6)")
        self.assertEqual(tree.result(), 2)

if __name__ == "__main__":
    unittest.main()

File: tree.cs
using System;
using NUnit.Framework;

public class Node
{
    private string operatorSymbol;
    private double? value;
    private Node left;
    private Node right;

    public Node(string operatorSymbol, double? value, Node left, Node right)
    {
        this.operatorSymbol = operatorSymbol;
        this.value = value;
        this.left = left;
        this.right = right;
    }

    public double Result()
    {
        switch (operatorSymbol)
        {
            case "x":
                return left.Result() * right.Result();
            case "÷":
                return left.Result() / right.Result();
            case "+":
                return left.Result() + right.Result();
            case "-":
                return left.Result() - right.Result();
            default:
                return value.HasValue ? value.Value : 0;
        }
    }

    public override string ToString()
    {
        switch (operatorSymbol)
        {
            case "x":
                return $"({left.ToString()} x {right.ToString()})";
            case "÷":
                return $"({left.ToString()} ÷ {right.ToString()})";
            case "+":
                return $"({left.ToString()} + {right.ToString()})";
            case "-":
                return $"({left.ToString()} - {right.ToString()})";
            default:
                return value.HasValue ? value.Value.ToString() : "0";
        }
    }
}

[TestFixture]
public class NodeTests
{
    [Test]
    public void TestNode()
    {
        // Create the tree structure
        Node tree = new Node(
            "÷",
            null,
            new Node(
                "+",
                null,
                new Node("", 7, null, null),
                new Node(
                    "x",
                    null,
                    new Node("-", null, new Node("", 3, null, null), new Node("", 2, null, null)),
                    new Node("", 5, null, null)
                )
            ),
            new Node("", 6, null, null)
        );

        // Assert that the string representation matches
        Assert.AreEqual("((7 + ((3 - 2) x 5)) ÷ 6)", tree.ToString());

        // Assert that the result is correct
        Assert.AreEqual(2, tree.Result());
    }
}

